src/xcode/ENA/ENA/Source/Models/Exposure/ExposureManager.swift

```diff
@@ -220,14 +221,37 @@ final class ENAExposureManager: NSObject, ExposureManager {
 	/// Needs to be called before `ExposureManager.enable()`
 	func activate(completion: @escaping CompletionHandler) {
 		Log.info("Trying to activate ENManager.")
+		
+		var activated = false
+		var retries = 0
+		let timeout = 5
+		
 		manager.activate { activationError in
 			if let activationError = activationError {
 				Log.error("Failed to activate ENManager: \(activationError.localizedDescription)", log: .api)
 				self.handleENError(error: activationError, completion: completion)
 				return
 			}
-			Log.info("Activated ENManager succesfully.")
-			completion(nil)
+			activated = true
+		}
+		
+		// Sometimes the ENF is broken. So we wait 5 seconds until we proceed with a deactivated ENF and log an error. Mostly, the ENF is activated instantly, so 5 seconds should be enough time to wait.
+		Timer.scheduledTimer(
+			withTimeInterval: 1,
+			repeats: true
+		) { timer in
+			if activated {
+				Log.info("Activated ENF within \(retries) seconds successfully.")
+				timer.invalidate()
+				completion(nil)
+			} else if retries >= timeout {
+				timer.invalidate()
+				Log.error("Could not activate ENF within \(retries) seconds. Proceed with deactivated ENF")
+				completion(ExposureNotificationError.notResponding)
+			} else {
+				Log.warning("Could not activate ENF within \(retries) seconds. Retry in 1 second.")
+				retries += 1
+			}
```

Comment:
A solution to not check just once every second could be to make `activated` a publisher and use [timeout](https://developer.apple.com/documentation/combine/fail/timeout(_:scheduler:options:customerror:)) to wait x seconds max for it to publish. That way the execution would continue immediately.

Or an even simpler solution could be to just keep the old `completion(nil)` where it was and just use the timer to check once after x seconds if it was called and if not, call `completion(ExposureNotificationError.notResponding)`

Reply:
Good idea (last one), then i can also use GDC for the timer.